tensorboard/webapp/widgets/data_table/header_cell_component.scss

```diff
@@ -17,77 +17,76 @@ limitations under the License.
 @import 'tensorboard/webapp/theme/tb_theme';
 
 $_accent: map-get(mat.get-color-config($tb-theme), accent);
+$_icon_size: 12px;
 
 :host {
   display: table-cell;
   padding: 4px;
   vertical-align: bottom;
+
+  &:hover {
+    .show-on-hover {
+      opacity: 0.3;
+    }
+  }
 }
+
 .sorting-icon-container {
-  width: 12px;
-  height: 12px;
+  width: $_icon_size;
+  height: $_icon_size;
   border-radius: 5px;
 }
-.cell mat-icon {
-  height: 12px;
-  width: 12px;
-}
+
 .cell {
   align-items: center;
   display: flex;
-}
-.cell .sorting-icon-container {
-  ::ng-deep path {
-    fill: unset;
-  }
-}
 
-.cell:hover .delete-icon-container {
-  opacity: 1;
-}
+  &:hover {
+    .delete-button-container {
+      opacity: 1;
+    }
+  }
 
-.delete-icon {
-  background-color: #fff;
-  border: 0;
-  border-radius: 50%;
-  color: mat.get-color-from-palette(mat.$gray-palette, 500);
-  font-size: 11px;
-  height: 11px;
-  margin-top: 2px;
-  padding: 0;
-  width: 11px;
+  mat-icon {
+    height: $_icon_size;
+    width: $_icon_size;
+  }
 
-  .mat-icon {
-    width: 100%;
-    height: 100%;
+  .sorting-icon-container {
+    ::ng-deep path {
+      fill: unset;
+    }
   }
 }
 
-.delete-icon:hover {
-  background-color: mat.get-color-from-palette(mat.$gray-palette, 400);
+.delete-button {
+  display: flex;
+  align-items: center;
   color: #fff;
+  font-size: $_icon_size;
+  height: $_icon_size;
+  width: $_icon_size;
+  line-height: $_icon_size;
   cursor: pointer;
-}
 
-.delete-icon-container {
-  margin-left: -12px;
-  opacity: 0;
-  position: absolute;
+  .mat-icon {
+    line-height: $_icon_size;
+  }
 }
 
-.delete-icon:hover {
+.delete-button-container {
+  border-radius: 50%;
   background-color: mat.get-color-from-palette(mat.$gray-palette, 400);
-  cursor: pointer;
+  margin-left: -$_icon_size;
```

Comment:
Just leaving a note here that this causes a problem where this icon appears on top of the column to the left. This was already an issue so it is not new here.

Reply:
Yeah... I didn't want to spend time experimenting with different styles for this right now but I agree that the placement is less than perfect